state-synchronizer/src/state_synchronizer.rs

```diff
@@ -268,16 +296,31 @@ impl StateSynchronizerClient {
         }
     }
 
-    /// Returns information about StateSynchronizer internal state. This should only
+    /// Returns information about the state sync internal state. This should only
     /// be used by tests.
     #[cfg(test)]
     pub fn get_state(&self) -> impl Future<Output = Result<SynchronizationState>> {
         let mut sender = self.coordinator_sender.clone();
         let (cb_sender, cb_receiver) = oneshot::channel();
+
+        async move {
+            sender
+                .send(CoordinatorMessage::GetSyncState(cb_sender))
+                .await?;
+            Ok(cb_receiver.await?)
```

Comment:
Yeah, unfortunately this is a no-go because there's no automatic conversion (i.e.,  std::ops::Try is not implemented for Synchronization state). We get this for free in the others, because they return `()` which implements these traits, but it's probably overkill to implement this here.

Reply:
you can use cb_receiver.await.map_err(|e| e.into)